### PR TITLE
Issue 124 add java nio path support

### PR DESCRIPTION
**Pull request by rmcdouga**
PR supporting Issue #124

### CODE REVIEW

**Comment by rmcdouga** [comment]
I see the builds are failing due to Javadoc.  I will correct this.  Also, it seems like the eca issue is related to the multiple email addresses I have on my GitHub account.  I have accepted the ECA but the automated test seems to be confused.  I will see what can be done.

File: api/src/main/java/jakarta/activation/FileDataSource.java [hunk 1]
```diff
@@ -10,129 +10,157 @@
...
+	public FileDataSource(File file) {
+		_path = file.toPath(); // save the file as Path object...
+	}
+
+	/**
+	 * Creates a FileDataSource from a File object. <i>Note: The file will not
+	 * actually be opened until a method is called that requires the file to be
+	 * opened.</i>
+	 *
+	 * @param file the file
```
Comment (lukasj) [inline api/src/main/java/jakarta/activation/FileDataSource.java:71]: param `file` does not match `path` argument in the constructor

**Review by lukasj — CHANGES_REQUESTED** [review]
in general, looks good, yet some fine-tunning is needed:

* avoid using tabs and stick with current formatting
* reformatting comments as part of the change does not make reviewer's life easier
* copyright year should be updated
* change should be noted in the spec document, at least in [the changelog](https://github.com/jakartaee/jaf-api/blob/master/spec/src/main/asciidoc/activation.adoc?plain=1#L878)

should marking `File` variants as deprecated be considered or do you propose to just keep them as they are for now?

File: api/src/main/java/jakarta/activation/MimetypesFileTypeMap.java [hunk 2]
```diff
@@ -355,6 +360,18 @@ public String getContentType(File f) {
         return this.getContentType(f.getName());
     }
 
+    /**
+     * Return the MIME type of the {@Code Path} object.
+     * The implementation in this class calls
+     * <code>getContentType(p.getFileName().toString())</code>.
+     *
+     * @param p the file {@Code Path}
```
Comment (lukasj) [inline api/src/main/java/jakarta/activation/MimetypesFileTypeMap.java:368]: tag names start with lower-case letter

**Comment by rmcdouga** [comment]
> avoid using tabs and stick with current formatting
Should I go back and change it back to what it was.  Sorry, the formatting style seemed very old-school to me, so I thought I was making things better but I understand the value of consistency.  I am happy to change it back.

> reformatting comments as part of the change does not make reviewer's life easier
Understood.

> copyright year should be updated
Will do.

> change should be noted in the spec document, at least in [the changelog](https://github.com/jakartaee/jaf-api/blob/master/spec/src/main/asciidoc/activation.adoc?plain=1#L878)
Will do.

> should marking File variants as deprecated be considered or do you propose to just keep them as they are for now?
This is a tough call.  Personally, I would deprecate them however the JDK maintainers have seen fit not to deprecate File even though it has been superseded.  I decided to follow the JDK lead and since it is not deprecated in the JDK, to leave it undeprecated in this API.  As a compromise, I put a note in the Javadoc indicating that the Path version should be preferred.

**Comment by rmcdouga** [comment]
> avoid using tabs and stick with current formatting
Should I go back and change it back to what it was.  Sorry, the formatting style seemed very old-school to me, so I thought I was making things better but I understand the value of consistency.  I am happy to change it back.

> reformatting comments as part of the change does not make reviewer's life easier
Understood.

> copyright year should be updated
Will do.

> change should be noted in the spec document, at least in [the changelog](https://github.com/jakartaee/jaf-api/blob/master/spec/src/main/asciidoc/activation.adoc?plain=1#L878)
Will do.

> should marking File variants as deprecated be considered or do you propose to just keep them as they are for now?
This is a tough call.  Personally, I would deprecate them however the JDK maintainers have seen fit not to deprecate File even though it has been superseded.  I decided to follow the JDK lead and since it is not deprecated in the JDK, to leave it undeprecated in this API.  As a compromise, I put a note in the Javadoc referring users to the

**Comment by lukasj** [comment]
> > avoid using tabs and stick with current formatting
> 
> Should I go back and change it back to what it was. Sorry, the formatting style seemed very old-school to me, so I thought I was making things better but I understand the value of consistency. I am happy to change it back.

yes, please - avoid tabs for new stuff. The codebase should be updated in one go followed by the inclusion of the change in `.git-blame-ignore-revs`

wrt deprecation - OK, let's keep it as it is and do nothing in that area for now

wrt failing ECA check - last two commits were probably done through different username (...www...) which has not signedthe ECA; the first one seems to be fine (probably)... can you try to amend them?

**Review by lukasj — APPROVED** [review]
LGTM, thanks!

**Pull request merged**